esx_service/vmdk_ops.py

```diff
@@ -778,6 +778,7 @@ def disk_attach(vmdk_path, vm):
             msg=("Failed to add PVSCSI Controller: %s", ex.msg)
             return err(msg)
         # Find the controller just added
+        devices = vm.config.hardware.device
```

Comment:
So this fix seems to address a regressing introduced by the attempt to get pci_slot_number in lines 782-785.  Why it is even needed when the slot number is already known (it's in the `key` which was  set in line 761) ? it feels like 781-785 can be safely removed and the pci_slot_number simply set to `key`.